src/RevCPU.cc

```diff
@@ -2609,10 +2630,8 @@ void RevCPU::CheckBlockedThreads(){
 // ----------------------------------
 void RevCPU::SetupArgs(uint32_t ThreadIDToSetup, RevFeature* feature){
   auto Argv = Opts->GetArgv();
-  // setup argc
   Threads.at(ThreadIDToSetup)->GetRegFile()->SetX(RevReg::a0, Argv.size());
-  Threads.at(ThreadIDToSetup)->GetRegFile()->SetX(RevReg::a1, Mem->GetStackTop() + 60);
-  return;
+  Threads.at(ThreadIDToSetup)->GetRegFile()->SetX(RevReg::a1, Mem->GetStackTop()+60);
```

Comment:
You may want to also set the `gp` register to the value of `loader->GetSymbolAddr("__global_pointer$")`

Reply:
per our parallel discussion, this is already done in thread init